staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithinGroupUntil adds 1 to the group, starts f in a new goroutine and calls wg.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithinGroupUntil(stopCh <-chan struct{}, wg Group, f func(stopCh <-chan struct{})) {
```

Comment:
I was actually thinking to name these functions something like that but I was not sure if names of that length are ok. Done.